lib/fit4ruby/FitDefinitionFieldBase.rb

```diff
@@ -28,11 +28,13 @@ def FitDefinitionFieldBase::undefined_value(fit_type)
       entry[2]
     end
 
-    def set_type(fit_type)
-      idx = FIT_TYPE_DEFS.index { |x| x[0] == fit_type }
+    def set_type(field, value)
+      fit_type = field.opts[:array] ? 'byte' : field.type
```

Comment:
This is based on the "Table 4-6. FIT Base Types and Invalid Values" of the "Flexible & Interoperable Data Transfer (FIT) Protocol Rev 2.3" PDF.